src/MainWindow.vala

```diff
@@ -132,7 +120,24 @@ public class AppCenter.MainWindow : Gtk.ApplicationWindow {
     }
 
     construct {
+        settings = new GLib.Settings ("io.elementary.appcenter.settings");
+
+        int window_x, window_y;
+        var rect = Gtk.Allocation ();
+
+        settings.get ("window-position", "(ii)", out window_x, out window_y);
+        settings.get ("window-size", "(ii)", out rect.width, out rect.height);
+
+        if (window_x != -1 ||  window_y != -1) {
+            move (window_x, window_y);
+        }
+
+        if (settings.get_boolean ("window-maximized")) {
+            maximize ();
+        }
+
         icon_name = "system-software-install";
+        set_allocation (rect);
```

Comment:
Should use `resize ()` - gtk documentation states that `set_allocation ()` should not be called directly.  Note that if `resize ()` is used, then you need to use `get_size ()` when saving the size parameters, not `get_allocation ()`. I think one includes the shadow and one doesnt.

Reply:
From the previous testing we had done, using `get_size ()` and `resize ()` lead to discrepancies with the window size on restore sometimes so you'd get windows that grow or shrink every time you restore it

Reply:
Best to find out the reason why ...